### PR TITLE
eslint replaces jshint

### PR DESCRIPTION
**Pull request by ocke**
:koala:

eslint replaces jshint

/cc @zendesk/quokka
### References
- Jira link: 
### Risks
- None

### CODE REVIEW

**Comment by adammw** [comment]
:heart_eyes:

**Comment by ocke** [comment]
Also the maintainer of eslintrb is a great guy...

**Comment by adammw** [comment]
:+1: although I really hope that the maintainer will keep the gem up to date.
Depends what the rest of the team thinks re: putting it behind a feature flag to control rollout.

**Comment by tobico** [comment]
:+1:

**Comment by sandlerr** [comment]
we'll just need to test in staging, there will be plenty of edge cases but we don't quite have the volume of new apps being created to catch them before 100% rollout anyway IMO

File: lib/zendesk_apps_support/validations/source.rb [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
+        },
         # predefined globals:
-        predef: %w(_ console services helpers alert confirm window document self
-                   JSON Base64 clearInterval clearTimeout setInterval setTimeout
-                   require module exports top frames parent moment),
-
-        browser: true
+        globals: Hash[
+          %w(_ Base64 services helpers require module exports moment)
+        .map { |x| [x, 'true'] }]
```
Comment (sandlerr) [inline lib/zendesk_apps_support/validations/source.rb]: you can put a boolean here can't you?

**Comment by sandlerr** [comment]
:+1:

**Comment by xuebingli** [comment]
👍 looks legit

**Comment by ocke** [comment]
@danielbreves can you 👍  as well?

**Comment by danielbreves** [comment]
This will likely break updating private apps that include jshint commented rules, right? This may require some advance communication. cc @maximeprades @dpawluk @Eshen
